tasks/interactive-bots.md

```diff
@@ -0,0 +1,152 @@
+
+  a. sending a few messages starting with `@followup` and `@follow-up` from different streams
+
+  b. checking that the messages showed up in the `followup` stream, prepended by the sender email
+
+## Task 1 - Create a links bot
+
+Follow the tutorial below to create your first simple bot that sends private messages to users.
+
+1. Duplicate the `followup.py` file under `links.py`, to build on existing code.
+
+
+2. Change the class name to `LinksHandler` and the last line to `handler_class = LinksHandler`.
+
+
+3. Edit the comments appropriately:
+
+  a. `class LinksHandler(object)`
+  > This plugin facilitates creating a list of resources you want to save while using Zulip. It looks for messages starting with "@link" or "@resource".
+  > In this example, we send resources to private messages.
+
+
+  b.  `def usage(self)`
+  > This plugin will allow users to flag messages as being resources and store them in private messages with the bot. Users should preface messages with "@link" or "@resource".
+
+
+4. Edit the `triage_message`:
+
+  ```
+  if message['display_recipient'] == 'links':
+      return False
+  is_link = (original_content.startswith('@link') or
+             original_content.startswith('@resource'))
+
+  return is_link
+  ```
+
+  The bot now responds to `@link` and `@resource`.
+
+
+5. Edit the `handle_message`:
+
+  ```
+  if original_content.startswith('@link'):
+      new_content = original_content.replace('@link',
+                                         'from %s:' % (original_sender,))
+  else:
+      new_content = original_content.replace('@resource',
+                                         'from %s:' % (original_sender,))
+  ```
+
+6. Check that the bot is working as expected:
+
+  - it reacts to the `@link` and `@resource` at the beginning of the message
+
+  - it currently posts the message to the followup stream, as the followup bot would
+
+
+7. Edit the `handle_message`:
+
+  ```
+  client.send_message(dict(
+      type='private',
+      to=original_sender,
+      content=new_content,
+  ))
+  ```
+
+8. Check that the bot is working as expected:
+
+  - it reacts to the `@link` and `@resource` at the beginning of a message
+
+  - it currently sends a private message from the admint account to the author of the original message
+
+[TODO] figure out why some bot credentials don't work locally
+
+## Task 2 - Create your own bots
+
+Some ideas on what the bots could do:
+
+  - post to an issue tracker like GitHub
+
+  - post events to a calendar like Google Calendar
+
+  - send group messages to user groups
+
+  - initiate a timer and post a message after a specific time interval
+
+  - post answers on specific topics - similar to `man` in terminal
+
+For inspiration you can look through the [current issues](https://github.com/zulip/zulip/issues) or ask other users what they might find useful.
+
+[TODO] post additional ideas that are relevant to current issues
```

Comment:
I guess this one also needs a "completion" point -- maybe it should be that they've submitted a pull request to add their bot to Zulip?